near-lake-primitives/src/types/block.rs

```diff
@@ -0,0 +1,51 @@
+use super::receipts;
+use crate::near_indexer_primitives::{types::AccountId, CryptoHash, StreamerMessage};
+
+#[derive(Debug, Clone)]
+pub struct Block {
+    header: BlockHeader,
+    executed_receipts: Vec<receipts::ExecutedReceipt>,
+}
+
+impl Block {
+    pub fn header(&self) -> &BlockHeader {
+        &self.header
+    }
+
+    pub fn receipts(&self) -> &[receipts::ExecutedReceipt] {
+        &self.executed_receipts
+    }
+}
+
+impl From<&StreamerMessage> for Block {
+    fn from(streamer_message: &StreamerMessage) -> Self {
+        Self {
+            header: streamer_message.into(),
+            executed_receipts: streamer_message
+                .shards
+                .iter()
+                .flat_map(|shard| shard.receipt_execution_outcomes.iter())
+                .map(Into::into)
+                .collect(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct BlockHeader {
```

Comment:
Do I miss some useful data from the `BlockHeaderView` wdyt?

Reply:
I see the following important fields:

* timestamp_nanosec
* epoch_id
* next_epoch_id
* gas_price
* total_supply
* latest_protocol_version
* random_value
* chunks_included ?
* validator_proposals ?

Will there still be access to the "raw" block header?

Reply:
There is always access to the entire `StreamerMessage` of course.

I am going to add these fields I guess